src/toxcore/onion/packet/friend_request.rs

```diff
@@ -80,12 +90,12 @@ mod tests {
     // Test for overflow of to_bytes.
     #[test]
     fn friend_req_to_bytes_overflow() {
-        let large_string = String::from_utf8(vec![32u8; MAX_FRIEND_REQUEST_DATA_SIZE + 1]).unwrap();
+        let large_string = String::from_utf8(vec![32u8; MAX_FRIEND_REQUEST_MSG_SIZE + 1]).unwrap();
         let friend_req = FriendRequest {
             nospam: NoSpam::random(),
             msg: large_string
         };
-        let mut buf = [0; MAX_FRIEND_REQUEST_DATA_SIZE + 1]; // `1` is for packet_id.
+        let mut buf = [0; MAX_ONION_CLIENT_DATA_SIZE + 1]; // `1` is to provide enough space for success of serializing
```

Comment:
No need to make `+ 1`, because we will for sure serialize in into buf with size = MAX_ONION_CLIENT_DATA_SIZE

Reply:
https://github.com/tox-rs/tox/blob/ab27c8d11c4f02944f7ff5d30ce986ad4c56be49/src/toxcore/onion/packet/friend_request.rs#L93

My intention is to provide even if ser was success.

Reply:
Ah I see. Good job.